flang/lib/Lower/OpenMP.cpp

```diff
@@ -2411,6 +2451,10 @@ genTargetOp(Fortran::lower::AbstractConverter &converter,
   mlir::Value ifClauseOperand, deviceOperand, threadLimitOperand;
   mlir::UnitAttr nowaitAttr;
   llvm::SmallVector<mlir::Value> mapOperands;
+  llvm::SmallVector<mlir::Value> devicePtrOperands, deviceAddrOperands;
+  llvm::SmallVector<mlir::Type> useDeviceTypes;
+  llvm::SmallVector<mlir::Location> useDeviceLocs;
+  llvm::SmallVector<const Fortran::semantics::Symbol *> useDeviceSymbols;
```

Comment:
These variables are currently unused from what I can see.

For the use_device_ptr/addr clauses for the target data directive. These were used to create block_arguments for these variables, because we wanted to replace all of their uses with new values generated by OMPIRBuilder.

If we are looking for a similar behaviour here then, and we want to add them as block_arguments as well. Then probably this change would be easier once #67164 is merged as that would already have the base code for doing this.